autogen/src/main.rs

```diff
@@ -61,65 +62,82 @@ fn main() {
         panic!("SPIRV-Headers missing - please checkout using git submodule");
     }
 
-    let mut contents = String::new();
-
-    {
-        let path = autogen_src_dir
-            .join("external/SPIRV-Headers/include/spirv/unified1/spirv.core.grammar.json");
-        let mut file = fs::File::open(path).unwrap();
-        file.read_to_string(&mut contents).unwrap();
-    }
-
     let grammar: structs::Grammar = {
-        let mut original = serde_json::from_str(&contents).unwrap();
+        let mut original =
+            serde_json::from_str(
+                &std::str::from_utf8(
+                    &fs::read(autogen_src_dir.join(
+                        "external/SPIRV-Headers/include/spirv/unified1/spirv.core.grammar.json",
+                    ))
+                    .unwrap(),
+                )
+                .unwrap(),
+            )
+            .unwrap();
         map_reserved_instructions(&mut original);
         original
     };
 
-    // For GLSLstd450 extended instruction set.
-    {
-        let path = autogen_src_dir.join(
-            "external/SPIRV-Headers/include/spirv/unified1/extinst.glsl.std.450.grammar.json",
-        );
-        let mut file = fs::File::open(path).unwrap();
-        contents.clear();
-        file.read_to_string(&mut contents).unwrap();
-    }
-    let gl_grammar: structs::ExtInstSetGrammar = serde_json::from_str(&contents).unwrap();
-
-    // For OpenCL extended instruction set.
-    {
-        let path = autogen_src_dir.join(
-            "external/SPIRV-Headers/include/spirv/unified1/extinst.opencl.std.100.grammar.json",
-        );
-        let mut file = fs::File::open(path).unwrap();
-        contents.clear();
-        file.read_to_string(&mut contents).unwrap();
-    }
-    let cl_grammar: structs::ExtInstSetGrammar = serde_json::from_str(&contents).unwrap();
+    let extended_instruction_sets = [
+			("GLSL.std.450", "GLOp", "https://www.khronos.org/registry/spir-v/specs/unified1/GLSL.std.450.html"),
+			("OpenCL.std.100", "CLOp", "https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.ExtendedInstructionSet.100.html"),
+			("NonSemantic.DebugPrintF", "DebugPrintFOp", "https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/docs/debug_printf.md"),
```

Comment:
Please use spaces to indent this. The autoformatter refuses to do so because the lines are too long.

```suggestion
        ("GLSL.std.450", "GLOp", "https://www.khronos.org/registry/spir-v/specs/unified1/GLSL.std.450.html"),
        ("OpenCL.std.100", "CLOp", "https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.ExtendedInstructionSet.100.html"),
        ("NonSemantic.DebugPrintF", "DebugPrintFOp", "https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/docs/debug_printf.md"),
```